HPS/Heartland/view/adminhtml/templates/form/cc.phtml

```diff
@@ -1,151 +1,285 @@
+                            return true;
+                        },
+
+                        hpsBusy: function () {
+                            $("#checkout-loader-iframeEdition").fadeIn();
+                        },
+
+                        hpsNotBusy: function () {
+                            $("#checkout-loader-iframeEdition").fadeOut();
+                            _HPS_EnablePlaceOrder()
+                        },
+
+                        hpsShowCcForm: function (publicKey) {
+                            if (publicKey) {
+
+                                $("#iframes").fadeIn();
+                                HPS_SecureSubmit($,document, Heartland, publicKey);
+
+
+                            }
+
+                        }
+                        ,
+                        hideNewCardForm: function () {
+                            $("#iframes").fadeOut();
+                        },
+
+                        getCode: function () {
+                            try {
+                                $("#iframes").fadeOut();
+                            } catch (e) {
+                            }
+                            return '<?php /* @noEscape */ echo $code; ?>';
+
+                        },
+
+                        isActive: function () {
+                            return true;
+                        },
+
+
+
+
+                        }
+                });
+
+
+            }
+        );
+
+
+    </script>
+    <table id="SavedCardsTable">
+        <caption>Stored Payments for this customer(if any)</caption>
+        <tbody>
+        <?php
+        foreach ($hpsTokens as $token) {
+            $token['token_value'] = $token['heartland_storedcard_id'];
+            unset($token['heartland_storedcard_id'], $token['dt'], $token['customer_id']);
+            $tokenJson = addslashes(json_encode($token));
+            //{"token_value":"3","cc_last4":"1111","cc_type":"visa","cc_exp_month":"2","cc_exp_year":"2025"}
+            $ccTypeHPS = strtoupper($token['cc_type']);
+            $jpsTokenTableRow = <<<TROW
+                <tr onclick='return {$code}._HPS_setHssTransaction1(JSON.parse("{$tokenJson}"));'>
+                    <td width="width:100px" ><input style="width:100px;cursor:pointer;" type="radio" name="HPSTokens[]" id="hssCardSelected{$token['token_value']}"></td>
+                    <th align="left" id="image_holder_{$token['token_value']}">{$ccTypeHPS} ending in {$token['cc_last4']}<br />Expiring on {$token['cc_exp_month']}/{$token['cc_exp_year']}<span class="card-type-{$token['cc_type']}" /></th>
+                    <td width="width:50px" ><input onclick="return {$code}.deleteCard({$token['token_value']},this
+                    .parentNode.parentNode)"
+                    style="color:red;font-size:
+                    2em;background-color: transparent;border: none" type="button"
+                     value="X"></td>
+                </tr>
+
+TROW;
+            echo $jpsTokenTableRow;
```

Comment:
We're in a template file, so building HTML within a PHP string shouldn't occur. This should be output directly as HTML with calls to PHP when necessary